Created OuttakeSequence.java

avakimc opened the pull request:
* moves lift to a given scoring position, opens the grabber, waits, and then closes the grabber
* added GRABBER_WAIT_TIME_SECONDS placeholder to Constants.Lift
* changed atPosition method in Lift from private to public

avakimc committed:
Created OuttakeSequence.java

* moves lift to a given scoring position, opens the grabber, waits, and then closes the grabber
* added GRABBER_WAIT_TIME_SECONDS placeholder to Constants.Lift
* changed atPosition method in Lift from private to public

balghane commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -136,6 +136,9 @@ public static final class Lift {
         Constants.PLACEHOLDER_DOUBLE;
+
+    /** The time in seconds for the grabber to open in OuttakeSequence*/
+    public static final double GRABBER_WAIT_TIME_SECONDS = PLACEHOLDER_DOUBLE;
```
outtake should be in the name somewhere

balghane replied:
also should be a calibration

balghane commented on src/main/java/frc/robot/commands/OuttakeSequence.java:
```diff
@@ -0,0 +1,41 @@
+  private Lift lift;
+  private LiftPosition scoringPosition;
+
+  public OuttakeSequence(Intake intake, Lift lift, LiftPosition scoringPosition) {
```
Instead of `scoringPosition` we should have some fixed outtake position. You'll probably need to add this to the Lift class, `LiftPosition` enum, the positions map, etc. It should be considered above start

balghane commented on src/main/java/frc/robot/commands/OuttakeSequence.java:
```diff
@@ -0,0 +1,41 @@
+    addCommands(liftToOuttakePos, openGrabber, waitForGrabber, closeGrabber);
+  }
+
+  /**
+   * trigger the lift to move to the desired scoring position (shelf, score_mid, score_high) ends
+   * when the lift reaches the desired position
+   */
+  private ParallelRaceGroup liftToOuttakePos =
+      new RunCommand(() -> lift.setDesiredPosition(scoringPosition), lift)
+          .until(() -> lift.atPosition(scoringPosition));
+
+  /** opens the grabber */
+  private InstantCommand openGrabber = new InstantCommand(lift::openGrabber, lift);
+
+  /** waits for the amount fo time for the grabber to open */
+  private WaitCommand waitForGrabber = new WaitCommand(Constants.Lift.GRABBER_WAIT_TIME_SECONDS);
+
+  /** closes the grabber */
+  private InstantCommand closeGrabber = new InstantCommand(lift::closeGrabber, lift);
```
It's usually nice when the member variables like this are at the start of the class definition, and not after any constructors or functions. Also I think we can just put these in one big `addCommands` call instead of saving them as member variables?

avakimc committed:
Revisions to OuttakeSequence.java and Finished some Lift Functions

* Revisions to OuttakeSequence.java
   - changed each command in the sequence from being a member variable to just being created in the addCommands() method
   - renamed grabber wait seconds to OUTTAKE_GRABBER_WAIT_SECONDS and moved it into Cal

* Skeleton Lift Functions
   - completed home() function
   - replaced getRidOfObject() function with our OuttakeSequence

balghane commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -115,8 +116,9 @@ public void burnFlashSparks() {
    * joysticks}.
    */
   private void configureBindings() {
+    OuttakeSequence o = new OuttakeSequence(lift);
```
Is this needed?

avakimc replied:
No, sorry. This was left in from when we were trying to figure out why OuttakeSequence() wasn't working.

balghane commented on src/main/java/frc/robot/commands/OuttakeSequence.java:
```diff
@@ -0,0 +1,22 @@
+
+public class OuttakeSequence extends SequentialCommandGroup {
+
+  /** moves lift to OUTTAKING position, opens the grabber, waits, closes the grabber */
```
This comment should be on the class, not the constructor

avakimc committed:
Final Revisions to OuttakeSequence.java

* removed unnecessary object of OuttakeSequence in RobotContainer.java
* moved description comment in OuttakeSequence.java from above the constructor to above the class

avakimc commented:
/format

github-actions[bot] committed:
Formatting fixes
